File: client/EditUserAccountWindow.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace repatriator_client
{
    public partial class EditUserAccountWindow : Form
    {
        public class UserAccount
        {
            public string username;
            public string password;
            public bool is_admin;
        }

        private UserAccount user_account = null;

        public EditUserAccountWindow()
        {
            InitializeComponent();
        }

        public UserAccount showGetNewUser(IWin32Window owner)
        {
            usernameTextBox.Text = "";
            passwordTextBox.Text = "";
            adminCheckBox.Checked = false;
            usernameTextBox.Focus();

            this.Text = "New User Account";
            this.ShowDialog(owner);

            return user_account;
        }

        public UserAccount showEditUser(IWin32Window owner, UserAccount current_account)
        {
            usernameTextBox.Text = current_account.username;
            passwordTextBox.Text = current_account.password;
            adminCheckBox.Checked = current_account.is_admin;
            usernameTextBox.Focus();

            this.Text = "Edit User Account";
            this.ShowDialog(owner);

            return user_account;
        }

        private void okButton_Click(object sender, EventArgs e)
        {
            this.user_account = new UserAccount();
            this.user_account.username = usernameTextBox.Text;
            this.user_account.password = passwordTextBox.Text;
            this.user_account.is_admin = adminCheckBox.Checked;
            this.Close();
        }

        private void cancelButton_Click(object sender, EventArgs e)
        {
            this.user_account = null;
            this.Close();
        }
    }
}

File: client/EditUserAccountWindow.cpp
#include "EditUserAccountWindow.h"
#include "ui_EditUserAccountWindow.h"

EditUserAccountWindow * EditUserAccountWindow::s_instance = NULL;

EditUserAccountWindow::EditUserAccountWindow(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::EditUserAccountWindow),
    m_account()
{
    ui->setupUi(this);

    bool success;
    success = connect(this, SIGNAL(accepted()), this, SLOT(handleAccepted()));
    Q_ASSERT(success);
    success = connect(this, SIGNAL(rejected()), this, SLOT(handleRejected()));
    Q_ASSERT(success);
}

EditUserAccountWindow::~EditUserAccountWindow()
{
    delete ui;
}

void EditUserAccountWindow::changeEvent(QEvent *e)
{
    QDialog::changeEvent(e);
    switch (e->type()) {
    case QEvent::LanguageChange:
        ui->retranslateUi(this);
        break;
    default:
        break;
    }
}

EditUserAccountWindow * EditUserAccountWindow::instance()
{
    if (! s_instance)
        s_instance = new EditUserAccountWindow();
    return s_instance;
}

QSharedPointer<EditUserAccountWindow::UserAccount> EditUserAccountWindow::showGetNewUser()
{
    m_account = QSharedPointer<UserAccount>(new UserAccount());

    // clear ui
    ui->usernameLineEdit->setText("");
    ui->passwordLineEdit->setText("");
    ui->adminCheckBox->setChecked(false);
    ui->usernameLineEdit->setFocus(Qt::OtherFocusReason);

    this->setWindowTitle(tr("New User Account"));
    this->exec();

    return m_account;
}

void EditUserAccountWindow::showEditUser(QSharedPointer<UserAccount> account)
{
    m_account = account;

    ui->usernameLineEdit->setText(m_account.data()->username);
    ui->passwordLineEdit->setText(m_account.data()->password);
    ui->adminCheckBox->setChecked(m_account.data()->is_admin);
    ui->usernameLineEdit->setFocus(Qt::OtherFocusReason);

    this->setWindowTitle(tr("Edit User Account"));
    this->exec();
}

void EditUserAccountWindow::handleAccepted()
{
    m_account.data()->username = ui->usernameLineEdit->text();
    m_account.data()->password = ui->passwordLineEdit->text();
    m_account.data()->is_admin = ui->adminCheckBox->isChecked();
}

void EditUserAccountWindow::handleRejected()
{
    m_account.clear();
    Q_ASSERT(m_account.isNull());
}
